src/documents/bulk_download.py

```diff
@@ -37,32 +44,38 @@ def _formatted_filepath(
         doc: Document,
         archive: bool = False,
         folder: str = "",
-    ):
+    ) -> Path:
         """
         Constructs a full file path for the given document to be used inside
         the zipfile.
 
         The path is already unique, as handled when a document is consumed or updated
         """
         if archive and doc.has_archive_version:
-            in_archive_path = os.path.join(folder, doc.archive_filename)
+            if doc.archive_filename is None:
```

Comment:
This line cannot be hit, `has_archive_version` checks the condition of `archive_filename is not None` already.

If this is just for typing, you could use `TYPE_CHECKING` and assert the filename is not None there

Reply:
That's a perfect solution!